.github/workflows/publish.yaml

```diff
@@ -0,0 +1,14 @@
+# A CI configuration to auto-publish pub packages.
+
+name: Publish
+
+on:
+  pull_request:
+    branches: [ master ]
```

Comment:
Doesn't this line cause publishing when a pull request is made to master branch? Without a tag pushed to the repo?

Reply:
Yeah, it's not straightforward to know what this will do just by looking at it; this series of lines will:
- execute the job below if a PR is opened which targets the master branch, or
- if a tag is pushed to the repo

If you dig into the script this references (https://github.com/dart-lang/ecosystem/blob/main/.github/workflows/publish.yaml), you can see a bit more detail. There, it runs our publishing tool in validation mode when its triggered from a PR, and in publish mode when its triggered from a push event.